lib/baseFactory.js

```diff
@@ -113,6 +112,31 @@ class BaseFactory {
     }
 
     /**
+     * Remove a Model
+     * @method remove
+     * @param  {Mixed}    config               Config object
+     * @return {Promise}
+     */
+    remove(config) {
```

Comment:
Right, this is good for when the model has no child relationships 👍 

Using your example of pipelines, the child relationships are attached to the object (i.e., [jobs](https://github.com/screwdriver-cd/models/blob/master/lib/pipeline.js#L130)). When constructing a model, it's not easy to construct an object in an asynchronous fashion. The factories help address that concern by using them as an intermediary while the respective model is loaded with data asynchronously. 

Once the model is loaded with data, we shouldn't need the factory for anything else; the remaining CRUD operations can be handled by the models themselves.